### PR TITLE
[PTD][BE][c10d] Add some code documents for TCPStore code and cosmetic changes to libUVStore code

### PR DESCRIPTION
**Pull request by fduwjj**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #130496

While designing something else when TCPStore is needed. I spent some time digging into the codebase of TCPStore and found that the code is a little bit challenging to understand without proper documents. Although people from OSS community must be smarter than me, I still want to document my findings in the code so that devs and users can use them as a reference down the road.

Also for libuv, we need to make private variables with a "_", so it's a pure renaming of private variables such as `tcpServer`.

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

File: torch/csrc/distributed/c10d/TCPStore.hpp [hunk 1]
```diff
@@ -9,6 +9,33 @@
 namespace c10d {
 namespace detail {
 
+// TCPStore essentially is a key-value pair storage we used for rendezvous
```
Comment (wconstab) [inline torch/csrc/distributed/c10d/TCPStore.hpp:12]: TCPStore is a key-value store used by PyTorch mainly for distributed rendezvous, but for other purposes as well.

File: torch/csrc/distributed/c10d/TCPStore.hpp [hunk 2]
```diff
@@ -9,6 +9,33 @@
...
+// synchronously.
+// 2. LibUVTCPStoreBackend: an event-driven asynchronous stream processing that
+// leverages libuv library (https://github.com/libuv/libuv) for better
+// performance. And this backend now is recommended to users. (We set the
+// default value of `useLibUV` inside `TCPStoreOptions` to true now, so users
+// should get it by default).
+//
+// Code structure:
+// ├── TCPStore client side API and server setup code:
+// │   TCPStore.hpp/TCPStore..cpp
```
Comment (wconstab) [inline torch/csrc/distributed/c10d/TCPStore.hpp:31]: duplicate ..

**Review by wconstab — APPROVED** [review]

File: torch/csrc/distributed/c10d/TCPStore.hpp (hunk 2 above)
Comment (fduwjj) [reply torch/csrc/distributed/c10d/TCPStore.hpp:31]: opps, Let me clean it up..

File: torch/csrc/distributed/c10d/TCPStore.hpp [hunk 3]
```diff
@@ -9,6 +9,33 @@
 namespace c10d {
 namespace detail {
 
+// TCPStore is a key-value store used by PyTorch mainly for distributed
```
Comment (wz337) [inline torch/csrc/distributed/c10d/TCPStore.hpp:12]: Appreciate great documentation. I am learning just by reading this nice doc.

**Comment by fduwjj** [comment]
Somehow this PR did not get merged... Let's merge it now.

**Pull request closed without merging**